hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/ContextTransactionProcessorTest.java

```diff
@@ -162,6 +179,90 @@ void stillChargesHapiFeesOnAbort() {
         verify(rootProxyWorldUpdater).commit();
     }
 
+    @Test
+    void stillChargesHapiFeesOnHevmException() {
+        final var contractsConfig = CONFIGURATION.getConfigData(ContractsConfig.class);
+        final var processors = Map.of(VERSION_046, processor);
+        final var subject = new ContextTransactionProcessor(
+                null,
+                context,
+                contractsConfig,
+                CONFIGURATION,
+                hederaEvmContext,
+                tracer,
+                rootProxyWorldUpdater,
+                hevmTransactionFactory,
+                feesOnlyUpdater,
+                processors,
+                customGasCharging);
+
+        given(context.body()).willReturn(transactionBody);
+        given(hevmTransactionFactory.fromHapiTransaction(transactionBody)).willReturn(HEVM_Exception);
+        given(transactionBody.transactionIDOrThrow()).willReturn(transactionID);
+        given(transactionID.accountIDOrThrow()).willReturn(SENDER_ID);
+
+        final var outcome = subject.call();
+
+        verify(rootProxyWorldUpdater).commit();
+        assertEquals(INVALID_CONTRACT_ID, outcome.status());
```

Comment:
I still have trouble understanding hapi tests.  Test is (by name) trying to confirm that fees are charged.  Exactly where does that check happen?  (Next test too.).

Reply:
These are unit tests?  Checking that the correct handling of `HederaEvmTransaction` with an exception. Next test checks to see if the correct type of `HederaEvmTransaction` gets created when an exception occurs.

Reply:
oh yeah, unit test, my bad.  but still my question is: name of test is `stillChargesHapiFeesOnHevmException` but I don't see a check in the test body that any fee was charged.  I see only a check that `commit` was called and that the status is as expected.  `commit` being called doesn't mean a fee was charged, or does it?